packages/arb-token-bridge-ui/src/hooks/useArbQueryParams.tsx

```diff
@@ -48,12 +48,16 @@ export const useArbQueryParams = () => {
 // Defined here so that components can directly rely on this for clean amount values and not rewrite parsing logic everywhere it gets used
 const AmountQueryParam = {
   // type of amount is always string | undefined coming from the input element onChange event `e.target.value`
-  encode: (amount: string | null | undefined) => amount,
+  encode: (amount: string | null | undefined) => {
+    const amountRegex = new RegExp(/^\d*(\.\d*)?$|MAX/i)
```

Comment:
NIT, but can we keep the amountRegex check and the MAX check separate and not part of the same regex? Separation of concerns ftw :)